roles/mirror-registry/tasks/main.yml

```diff
@@ -0,0 +1,46 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Make directory for mirror-registry to use
+  become: true
+  ansible.builtin.file:
+     path: "{{ cifw_mirror_registry_base_dir }}/registry"
+     owner: "{{ lookup('env', 'USER') }}"
+     group: "{{ lookup('env', 'USER') }}"
+     state: directory
+
+- name: Download mirror-registry tools
+  ansible.builtin.get_url:
+     url: https://github.com/quay/mirror-registry/releases/download/v1.3.10/mirror-registry-online.tar.gz
+     dest: "{{ cifw_mirror_registry_base_dir }}"
+
+- name: Untar mirror-registry tools
+  ansible.builtin.unarchive:
+    src: "{{ cifw_mirror_registry_base_dir }}/mirror-registry-online.tar.gz"
+    dest: "{{ cifw_mirror_registry_base_dir }}"
+
+- name: Add hosts entry in /etc/hosts for mirror-registry
+  become: true
+  ansible.builtin.lineinfile:
+    dest: /etc/hosts
+    line: "127.0.0.1 {{ cifw_mirror_registry_fqdn }}"
+    state: present
+
+- name: Install mirror-registry
+  become: true
+  ansible.builtin.shell: |
+    {{ cifw_mirror_registry_base_dir }}/mirror-registry install -v --quayHostname {{ cifw_mirror_registry_fqdn }} --quayRoot {{ cifw_mirror_registry_base_dir }}/registry --initPassword {{ cifw_mirror_registry_init_password }}
```

Comment:
We need to decide on a good way to derive this password. Ultimately, it would be nice to derive it from something else we're already using to avoid having multiple required secrets.

Reply:
can't the ci-token or pull-secret data be used here? Though we probably have to get "personal" credentials.. Guess it's in order to auth against Quay to bypass any threshold?

Also, seeing that command, we may want to use our `ci_script` instead, to get nice logging.

Reply:
I could probably cut the last 10 chars from the ci-token.

Can I always expect that the ci-token will be at the path defined by `cifmw_manage_secrets_citoken_dest`?
```
# CI Token management
cifmw_manage_secrets_citoken_dest: >-
  {{ (cifmw_manage_secrets_basedir,
      'secrets', 'ci_token') | path_join
  }}

```

Reply:
So, I decided it's probably easier to just randomly generate a string that will be valid for the duration of the Ansible execution. It has no persistence outside of the execution, so the only thing would be that we need to ensure that fact is set at the right time during the task execution workflow:
https://github.com/openstack-k8s-operators/ci-framework/pull/1297/files#diff-e9fb47ac2a1b864acec9fbe6bee0edb7d1a9f9c6abbdb61b8e3fd13ee5bc2e8eR22